### PR TITLE
Iter9

### PR DESCRIPTION
**Pull request by kripsy**

### CODE REVIEW

File: cmd/client/main.go [hunk 1]
```diff
@@ -32,13 +36,21 @@ func main() {
...
+	request, err := http.NewRequest(http.MethodPost, endpoint, &cb)
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		os.Exit(1)
 	}
 	// в заголовках запроса указываем кодировку
-	request.Header.Add("Content-Type", "application/x-www-form-urlencoded")
+	request.Header.Add("Content-Type", "application/json")
+
+	request.Header.Add("Accept-Encoding", "gzip")
```
Comment (arttet) [inline cmd/client/main.go]: Please move to another function

File: cmd/client/main.go [hunk 2]
```diff
@@ -54,6 +66,33 @@ func main() {
 		fmt.Fprintf(os.Stderr, "error: %v\n", err)
 		os.Exit(1)
 	}
+
 	// и печатаем его
 	fmt.Println(string(body))
+	fmt.Println((decompress(string(body))))
+}
+
+func compress(data string) bytes.Buffer {
```
Comment (arttet) [inline cmd/client/main.go]: It should not be in `main`

File: internal/app/handlers/handlers_test.go [hunk 3]
```diff
@@ -145,3 +156,84 @@ func TestGetURLHandler(t *testing.T) {
 		})
 	}
 }
+
+func TestAPIHandler_SaveURLJSONHandler(t *testing.T) {
```
Comment (arttet) [inline internal/app/handlers/handlers_test.go]: Please rename it.

File: internal/app/logger/logger.go [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+)
+
+var Log *zap.Logger = zap.NewNop()
+
+func InitLog(level string) error {
+
+	lvl, err := zap.ParseAtomicLevel(level)
+
+	if err != nil {
+		Log.Fatal("can't parse lovel zap logger", zap.String("message", err.Error()))
```
Comment (arttet) [inline internal/app/logger/logger.go]: `Failed to parse zap logger level`

File: internal/app/logger/logger.go [hunk 5]
```diff
@@ -0,0 +1,33 @@
+package logger
+
+import (
+	"go.uber.org/zap"
+)
+
+var Log *zap.Logger = zap.NewNop()
```
Comment (arttet) [inline internal/app/logger/logger.go]: Please don't use global variables

File: internal/app/middleware/gzip.go [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+func (c *compressWriter) Header() http.Header {
+	return c.w.Header()
+}
+
+func (c *compressWriter) Write(p []byte) (int, error) {
+	return c.zw.Write(p)
+}
+
+func (c *compressWriter) WriteHeader(statusCode int) {
+	if statusCode < 300 {
```
Comment (arttet) [inline internal/app/middleware/gzip.go:32]: This is a magic number

File: internal/app/storage/file_storage.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
...
+	"encoding/json"
+	"fmt"
+	"os"
+
+	"github.com/google/uuid"
+	"github.com/kripsy/shortener/internal/app/logger"
+	"go.uber.org/zap"
+)
+
+var FILENAME string
```
Comment (arttet) [inline internal/app/storage/file_storage.go]: Please don't use global variables

File: internal/app/storage/storage.go [hunk 8]
```diff
@@ -3,30 +3,50 @@ package storage
...
 		val, err := utils.CreateShortURL()
 		if err != nil {
 			return "", err
 		}
-		m.storage[string(url)] = val
+		m.storage[url] = val
+
+		e := make([]Event, 1)
+
+		e[0] = *NewEvent(val, url)
```
Comment (arttet) [inline internal/app/storage/storage.go]: What if we get `nil`?

**Review by arttet — APPROVED** [review]

File: cmd/client/main.go (hunk 1 above)
Comment (kripsy) [reply cmd/client/main.go]: done!

File: cmd/client/main.go (hunk 2 above)
Comment (kripsy) [reply cmd/client/main.go]: done!

File: internal/app/handlers/handlers_test.go (hunk 3 above)
Comment (kripsy) [reply internal/app/handlers/handlers_test.go]: done!

File: internal/app/logger/logger.go (hunk 4 above)
Comment (kripsy) [reply internal/app/logger/logger.go]: done!

File: internal/app/logger/logger.go (hunk 5 above)
Comment (kripsy) [reply internal/app/logger/logger.go]: done!

File: internal/app/storage/file_storage.go (hunk 7 above)
Comment (kripsy) [reply internal/app/storage/file_storage.go]: done!

File: internal/app/storage/storage.go (hunk 8 above)
Comment (kripsy) [reply internal/app/storage/storage.go]: done!

File: internal/app/middleware/gzip.go (hunk 6 above)
Comment (kripsy) [reply internal/app/middleware/gzip.go:32]: we compress data only with status code 201 at the moment)))
it's from theory example)

**Pull request merged**